Unattended upgrades api docs (#2484)
canonical/ubuntu-pro-client: docs/explanations/how_to_interpret_unattended_upgrades_is_running.md

> # How to interpret if unattended upgrades is running

Reviewer: I find this title to be slightly confusing. I'm not sure if it should be "How to tell if unattended-upgrades is running" or "How to interpret the output of unattended-upgrades".

Author: I think the latter should be better. Changing it now

> On Pro Client version 27.14~, we have introduced the `u.pro.unattended_upgrades.status.v1` endpoint.

Reviewer: ```suggestion
On Pro Client version 27.14~, we introduced the `u.pro.unattended_upgrades.status.v1` endpoint.
```

This endpoint is designed to provide users with an overview of the configuration and setup for

> unattended-upgrades on the machine. The expected output follows this json example:

Reviewer: ```suggestion
unattended-upgrades on the machine. The expected output follows this JSON example:
```

        "${distro_id}ESMApps:${distro_codename}-apps-security",
        "${distro_id}ESM:${distro_codename}-infra-security"
      ],
      "unattended_upgrades_disabled_reason": null,
      "unattended_upgrades_frequency_days": 1,
      "unattended_upgrades_last_run": null,
      "unattended_upgrades_running": true
    },
    "meta": {
      "environment_vars": [],
      "raw_config": {
        "APT::Periodic::Enable": "1",
        "APT::Periodic::Unattended-Upgrade": "1",
        "APT::Periodic::Update-Package-Lists": "1",
        "Unattended-Upgrade::Allowed-Origins": [
          "${distro_id}:${distro_codename}",
          "${distro_id}:${distro_codename}-security",
          "${distro_id}ESMApps:${distro_codename}-apps-security",
          "${distro_id}ESM:${distro_codename}-infra-security"
        ]
      }
    },
    "type": "UnattendedUpgradesStatus"
  },
  "errors": [],
  "result": "success",
  "version": "27.14~16.04.1",
  "warnings": []
}
```

As we can see from this output, we have a variable named `unattended_upgrades_running`. That variable
indicates if unattended-upgrades is properly configured and running on the machine.

> That value of this field will only be `true` if *ALL*  of the following prerequisites are also true:

Reviewer: ```suggestion
The value of this field will only be `true` if *ALL*  of the following prerequisites are also true:
```

* *`apt_periodic_job_enable` is true*: That variable indicates if the APT::Periodic::Enable configuration variable
  is turned on. If it is turned off, unattended-upgrades will not automatically run on the machine.

> * *`package_lists_refresh_frequency_days` is non-zero*:  That variable shows the value of APT::Periodic::Package-List-Frequency.

Reviewer: ```suggestion
* *`package_lists_refresh_frequency_days` is non-zero*: That variable shows the value of APT::Periodic::Package-List-Frequency.
```

  This configuration defines the daily frequency for updating package sources in the background. If it has a zero value, this step will never

>   happen and unattended-upgrades might not be able to install new version of the packages.

Reviewer: ```suggestion
  happen and unattended-upgrades might not be able to install new versions of the packages.
```

> * *`systemd_apt_timer_enabled` is true*: That variable is true if both `apt-daily.timer` and `apt-daily-upgrade.timer` are running

Reviewer: ```suggestion
* *`systemd_apt_timer_enabled` is true*: This variable is true if both `apt-daily.timer` and `apt-daily-upgrade.timer` are running
```

>   in the machine. These timers are the ones that control when unattended-upgrades run. The first job, `apt-daily.timer` is responsible

Reviewer: ```suggestion
  on the machine. These timers are the ones that control when unattended-upgrades run. The first job, `apt-daily.timer` is responsible
```

>   for triggering the package source updates while `apt-daily-upgrade.timer` is responsible for running unattended-upgrades to download

Reviewer: "the package source updates" is a bit ambiguous - I'm not sure if you're referring to a package called "source updates" (in which case, I think it should probably be decorated as inline code), or if it's referring to source updates to the package. Please clarify :)

Author: Good catch, I have updated the description. Let me know if it is better now

Reviewer: Perfect, thank you :)

  the packages. If one of these jobs is disabled, unattended-upgrades might not work as expected.

> * *`unattended_upgrades_allowed_origins` is not empty*: That variable define the origins that

Reviewer: ```suggestion
* *`unattended_upgrades_allowed_origins` is not empty*: This variable defines the origins that
```

Reviewer: Also, I'm not sure what origins means in this context. E.g. what's the expected/allowed format?

Author: We are displaying the format of those origins on the example we are providing. But let me know if I should further explain that

Reviewer: I think it should be fine as is, I just wanted to clarify

  unattended-upgrades can use to install a package. If that list is empty, no packages can be
  installed and unattended-upgrades will not work as expected.
* *`unattended_upgrades_frequency_days` is non-zero*: That variable shows the value of
  APT::Periodic::Unattended-Upgrade. This configuration defines the daily frequency for running
  unattended-upgrades in the background. Therefore, if it has a zero value, the command will never
  run.

> If any of those conditions are not meet, the variable

Reviewer: ```suggestion
If any of those conditions are not met, the variable
```
